sycl/test/basic_tests/accessor/accessor.cpp

```diff
@@ -1,8 +1,13 @@
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -Dsimplification_test -std=c++17 %s -o %t.s.out
```

Comment:
I suggest adding a separate test for simplified API.

Reply:
That was exactly, how it was done in the begging, but was reworked by recommendation from @sergey-semenov in order to reduce code duplication.

Reply:
I have a couple of concerns:
- I don't think there will be a lot of duplication. Probably the test code should be reduced to reduce the duplicated part.
- Using the test like this doubles the execution time, which can't be improved by "parallel test execution".

Reply:
> Using the test like this doubles the execution time, which can't be improved by "parallel test execution".

What do you think of moving the source code to a separate shared file and including it from 2 different tests like we do in device-code-split tests? It should be the best of both worlds, unless there's some drawback to that I'm missing.

Reply:
In this case, I can split this test in a separate PR, which would follow shortly - adding additional functionality for accessors simplifications

Reply:
okay

Reply:
> > Using the test like this doubles the execution time, which can't be improved by "parallel test execution".
> 
> What do you think of moving the source code to a separate shared file and including it from 2 different tests like we do in device-code-split tests? It should be the best of both worlds, unless there's some drawback to that I'm missing.

Sounds good to me.